runtime/lib/app-runtime.js

```diff
@@ -539,10 +539,11 @@ AppRuntime.prototype.stopMonologue = function (appId) {
  * @param {object} nlp
  * @param {object} action
  * @param {object} [options]
+ * @param {'voice' | 'text' | string} [options.source] - voice command source
```

Comment:
Wondering if it's appropriate to just add an additional option to distinguish if the voice command derived from actual voice context. Since `onVoiceCommand` would be called with nlp from any source like MQTT, DBUS, etc.

Reply:
I think we only have the following 3 ways to send voice command: speech(voice), text(asr) and mock via MQTT or others.